### PR TITLE
Add rule S6509: Extra boolean casts should be removed

### LINKED ISSUE #3709: Add rule S6509: Extra boolean casts should be removed
- https://eslint.org/docs/latest/rules/no-extra-boolean-cast
- requires new RSPEC

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
Fixes #3709

### CODE REVIEW

File: tests/linting/eslint/linter/fixtures/wrapper/quickfixes/no-extra-boolean-cast.js [hunk 1]
```diff
@@ -0,0 +1 @@
+if (!!x) {}
```
Comment (ilia-kebets-sonarsource) [inline tests/linting/eslint/linter/fixtures/wrapper/quickfixes/no-extra-boolean-cast.js:1]: what is the use of this?

File: src/linting/eslint/linter/quickfixes/messages.ts [hunk 2]
```diff
@@ -27,6 +27,7 @@
 const quickFixMessages = new Map<string, string>([
   ['comma-dangle', 'Remove this trailing comma'],
   ['eol-last', 'Add a new line at the end of file'],
+  ['no-extra-boolean-cast', 'Remove extra cast'],
```
Comment (ilia-kebets-sonarsource) [inline src/linting/eslint/linter/quickfixes/messages.ts:30]: and this?

**Review by ilia-kebets-sonarsource — APPROVED** [review]
see comments, otherwise LGTM!

File: src/linting/eslint/linter/quickfixes/messages.ts (hunk 2 above)
Comment (yassin-kammoun-sonarsource) [reply src/linting/eslint/linter/quickfixes/messages.ts:30]: ESLint has two types of "quick fixes": suggestions and fixes. While suggestions include a message for a quick fix, fixes don't. Therefore, we need to define ourselves the message of the fix.

**Pull request merged**